### PR TITLE
Update v23.1.28 release notes

### PR DESCRIPTION
**Pull request by mdlinville**
- Remove duplicate note
- Move security note to the correct part of the file
- In security note, add a link to a command a fix a link pointing to the wrong reference

### CODE REVIEW

**Review by RaduBerinde — APPROVED** [review]
Thank you very much!!

**Review by florence-crl — COMMENTED** [review]
lgtm, thanks for getting to the bottom of this.

**Review by florence-crl — APPROVED** [review]

File: src/current/_includes/releases/v23.1/v23.1.28.md [hunk 1]
```diff
@@ -3,6 +3,11 @@
 Release Date: October 10, 2024
 
 {% include releases/new-release-downloads-docker-image.md release=include.release %}
+
+<h3 id="v23-1-28-security-changes">Security changes</h3>
+
+ [`SHOW CHANGEFEED JOB`]({% link v23.1/show-changefeed-job.md %}), [`SHOW CHANGEFEED JOBS`]({% link v23.1/show-changefeed-jobs.md %}#show-changefeed-jobs), and [`SHOW JOBS`]({% link v23.1/show-jobs.md %}) commands no longer expose user sensitive information like `client_key`. [#129910][#129910]
```
Comment (florence-crl) [inline src/current/_includes/releases/v23.1/v23.1.28.md:9]: there is no page `v23.1/show-changefeed-job.md`, please remove link.

**Review by florence-crl — APPROVED** [review]
lgtm pending link removal

File: src/current/_includes/releases/v23.1/v23.1.28.md (hunk 1 above)
Comment (mdlinville) [reply src/current/_includes/releases/v23.1/v23.1.28.md]: Fixed, Kathryn asked me to link both commands to the same section.

**Pull request merged**